File: solutions/java/01-jm1/code/src/main/java/Main.java
import java.io.IOException;
import java.net.ServerSocket;

public class Main {
  public static void main(String[] args){
    ServerSocket serverSocket = null;
    try {
      serverSocket = new ServerSocket(6379);
      serverSocket.setReuseAddress(true);
      serverSocket.accept(); // Wait for connection
    } catch (IOException e) {
      System.out.println("IOException: " + e.getMessage());
    }
  }
}

File: solutions/zig/01-jm1/code/src/main.zig
const std = @import("std");

pub fn main(init: std.process.Init) !void {
    const io = init.io;

    const address = try std.Io.net.IpAddress.parseIp4("127.0.0.1", 6379);

    var server = try address.listen(io, .{
        .reuse_address = true,
    });
    defer server.deinit(io);

    const connection = try server.accept(io);
    defer connection.close(io);

    try std.Io.File.writeStreamingAll(std.Io.File.stdout(), io, "accepted new connection");
}
